### PR TITLE
Igor/complete exercise

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+                _setZnkExerciseSettings();
+
+                $ctrl.exerciseContent = exerciseContent;
+                $ctrl.exerciseResult = exerciseResult;
+                $ctrl._finishExercise = _finishExercise;
+            }
+
+            _init();
+            // if (angular.isUndefined(exerciseResult.startedTime)) {
```
Comment (odeds) [inline src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js]: needs the comment?

File: src/components/completeExercise/directives/completeExerciseExercise/completeExerciseExercise.directive.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                            $q.all([contentProm, titleProm, buttonFinishProm, buttonContinueProm]).then(function (results) {
+                                var content = results[0];
+                                var title = results[1];
+                                var buttonFinish = results[2];
+                                var buttonContinue = results[3];
+                                var timeOverPopupPromise = PopUpSrv.ErrorConfirmation(title, content, buttonFinish, buttonContinue).promise;
+
+                                timeOverPopupPromise.then(function () {
+                                    $ctrl.znkExercise._finishExercise();
+                                    // _finishExercise(exerciseResult);
```
Comment (odeds) [inline src/components/completeExercise/directives/completeExerciseExercise/completeExerciseExercise.directive.js]: remove comment

File: src/components/completeExercise/parser/completeExerciseTimer/completeExerciseTimer.parser.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    'use strict';
+
+    angular.module('znk.infra-web-app.completeExercise')
+        .directive('completeExerciseTimerParser', function completeExerciseTimer(){
+            var directive = {
+                restrict: 'A',
+                require: 'ngModel',
+                link: {
+                    pre: function(scope, element, attrs, ngModelCtrl){
+                        ngModelCtrl.$parsers.push(function(timeLeft){
```
Comment (odeds) [inline src/components/completeExercise/parser/completeExerciseTimer/completeExerciseTimer.parser.js]: maybe change to one base function

File: src/components/completeExercise/directives/completeExerciseProgressBar/completeExerciseProgressBar.directive.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    templateUrl: 'components/completeExercise/directives/completeExerciseProgressBar/completeExerciseProgressBarDirective.template.html',
+                    scope: {
+                        totalTime: '@',
+                        duration: '<'
+                    },
+                    link: function (scope, element) {
+                        var isAnimationSet, prevCls;
+                        var BAR_CLASSES = {
+                            'GREEN': 'green-state',
+                            'YELLOW': 'yellow-state',
```
Comment (odeds) [inline src/components/completeExercise/directives/completeExerciseProgressBar/completeExerciseProgressBar.directive.js]: it's should be yellow? not orange?

Comment (igormas) [inline src/components/completeExercise/directives/completeExerciseProgressBar/completeExerciseProgressBar.directive.js]: The color itself is orange, but it states should be same as traffic light.

File: src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js (hunk 1 above)
Comment (igormas) [inline src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js]: Fixed

File: src/components/completeExercise/directives/completeExerciseExercise/completeExerciseExercise.directive.js (hunk 2 above)
Comment (igormas) [inline src/components/completeExercise/directives/completeExerciseExercise/completeExerciseExercise.directive.js]: fixed

**Pull request merged**